oscarapi/middleware.py

```diff
@@ -79,13 +79,15 @@ def start_or_resume(session_id, session_type):
     return get_session(session_id, raise_on_create=True)
 
 
-def is_api_request(request):
-    path = request.path.lower()
-    api_root = reverse('api-root').lower()
-    return path.startswith(api_root)
+class IsApiRequest(object):
```

Comment:
and why not just add it as a method to HeaderSessionMiddleware?
you are always going to override that class, so no point in having a separate IsApiRequest class right?